lib/unifex/code_generators/common.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Unifex.CodeGenerators.Common do
+  import Unifex.CodeGenerator.Utils, only: [sigil_g: 2]
+  alias Unifex.CodeGenerator.BaseType
+
+  def get_ctx(specs) do
+    customized_types =
+      specs.structs
+      |> Enum.map(fn {struct_alias, struct_module_name, struct_fields} ->
+        {
+          struct_alias,
+          %{
+            struct_alias: struct_alias,
+            struct_module_name: struct_module_name,
+            struct_fields: struct_fields
+          }
+        }
+      end)
+      |> Enum.into(%{})
+
+    %{customized_types: customized_types}
+  end
+
+  def generate_struct_native_definition(
+        {struct_type_name, _struct_module_name, struct_fields},
+        code_generator,
+        ctx
+      ) do
+    struct_fields_definition =
+      struct_fields
+      |> Enum.map(fn {field_name, field_type} ->
+        BaseType.generate_declaration(field_type, field_name, code_generator, ctx)
+      end)
+      |> Enum.map(&Bunch.listify/1)
+      |> Enum.flat_map(fn x -> x end)
```

Comment:
Do we need these two lines?

Reply:
Yes, because some of the elements of mapped list are lists too (e.g. when type of field of struct is `:list`)